app/models/snp_reference.rb

```diff
@@ -3,4 +3,5 @@ class SnpReference < ActiveRecord::Base
   belongs_to :snp
   belongs_to :paper, polymorphic: true
   #validates_presence_of :snp, :paper
+  validates_uniqueness_of :snp_id, scope: [:paper_id, :paper_type]
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.